### PR TITLE
#2509: Fix for 'DoubleMath incorrectly rounds 0.99999999999999994 to 2 with rounding mode UP or CEILING

### PR DESCRIPTION
**Pull request by terzerm**
See Issue #2509

### CODE REVIEW

**Comment by terzerm** [comment]
I signed an individual CLA.

File: guava/src/com/google/common/math/DoubleMath.java [hunk 1]
```diff
@@ -66,14 +66,14 @@ static double roundIntermediate(double x, RoundingMode mode) {
         if (x >= 0.0 || isMathematicalInteger(x)) {
           return x;
         } else {
-          return x - 1.0;
+          return (long)x - 1.0;
```
Comment (eamonnmcmanus) [inline guava/src/com/google/common/math/DoubleMath.java]: I think this will do the wrong thing if x is too large to fit in a long.

Comment (terzerm) [inline guava/src/com/google/common/math/DoubleMath.java]: valid point --- should check that long boundaries are not breached.

weird that there's no test coverage for this --- cause i have run all DoubleMath tests.

Comment (lowasser) [inline guava/src/com/google/common/math/DoubleMath.java]: I concur, Eamonn: if x doesn't fit in a long, it's a mathematical integer anyway.

Comment (eamonnmcmanus) [inline guava/src/com/google/common/math/DoubleMath.java]: OK.

**Comment by cpovirk** [comment]
Fixed a little differently in 1b1163b7e2c121d4a5b25b8966714201551976c4. Thanks for reporting.

**Pull request closed without merging**